pyretailscience/plots/cohort.py

```diff
@@ -0,0 +1,96 @@
+"""This module provides functionality for creating cohort plots from pandas DataFrames.
+
+It is designed to visualize data distributions using color-coded heatmaps, helping to highlight trends and comparisons between different groups.
+
+### Core Features
+
+- **Color Mapping**: Uses a predefined colormap for visualizing data.
+- **Customizable Labels**: Supports custom labels for x-axis, y-axis, title, and colorbar.
+- **Source Text**: Provides an option to add source attribution to the plot.
+- **Grid and Tick Customization**: Applies standard styling for better readability.
+
+### Use Cases
+
+- **Cohort Analysis**: Visualize how different groups behave over time.
+- **Category-Based Heatmaps**: Compare values across different categories.
+
+### Limitations and Warnings
+
+- **Data Aggregation Required**: The module does not perform data aggregation; data should be pre-aggregated before being passed to the function.
+- **Fixed Color Mapping**: The module uses a predefined colormap without dynamic adjustments.
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from matplotlib import ticker
+from matplotlib.axes import Axes, SubplotBase
+
+import pyretailscience.style.graph_utils as gu
+from pyretailscience.style.tailwind import get_listed_cmap
+
+
+def plot(
+    df: pd.DataFrame,
+    cbarlabel: str,
+    x_label: str | None = None,
+    y_label: str | None = None,
+    title: str | None = None,
+    ax: Axes | None = None,
+    source_text: str | None = None,
+    percentage: bool = False,
+    **kwargs: dict,
+) -> SubplotBase:
+    """Plots a cohort plot for the given DataFrame.
+
+    Args:
+        df (pd.DataFrame): Dataframe containing cohort analysis data.
+        cbarlabel (str): Label for the colorbar.
+        x_label (str, optional): Label for x-axis.
+        y_label (str, optional): Label for y-axis.
+        title (str, optional): Title of the plot.
+        ax (Axes, optional): Matplotlib axes object to plot on.
+        source_text (str, optional): Additional source text annotation.
+        percentage (bool, optional): If True, displays cohort values as percentages. Defaults to False.
+        **kwargs: Additional keyword arguments for cohort styling.
+
+    Returns:
+        SubplotBase: The matplotlib axes object.
+    """
+    ax = ax or plt.gca()
+    cmap = get_listed_cmap("green")
+    im = ax.imshow(df, cmap=cmap, **kwargs)
+    cbar = ax.figure.colorbar(im, ax=ax)
+    cbar.ax.set_ylabel(cbarlabel, rotation=-90, va="bottom", fontsize="x-large")
+
+    ax.set_xticks(np.arange(df.shape[1]))
+    ax.set_yticks(np.arange(df.shape[0]))
+    ax.set_xticklabels(df.columns, rotation=-30, ha="right", rotation_mode="anchor")
```

Comment:
Can you make this so the numbers are not rotated please. Also replace the image
```suggestion
    ax.set_xticklabels(df.columns, ha="right", rotation_mode="anchor")
```

Reply:
@mayurkmmt Can you also remove the ha="right" so they're centered please.